rclcpp/test/test_logging.cpp

```diff
@@ -92,6 +92,14 @@ TEST_F(TestLoggingMacros, test_logging_named) {
   EXPECT_EQ("message 3", g_last_log_event.message);
 }
 
+TEST_F(TestLoggingMacros, test_logging_string) {
+  for (std::string i : {"one", "two", "three"}) {
+    RCLCPP_DEBUG(g_logger, "message " + i);
```

Comment:
I have added new tests, with more than one argument, but I understand that possible uses are:

- RCLCPP_DEBUG(g_logger, std::string("message " + i));
- RCLCPP_DEBUG(g_logger, std::string("message " "one"));
- RCLCPP_DEBUG(g_logger, std::string("message %d"), 1); //This works, but I am not sure if it is correct
- RCLCPP_DEBUG(g_logger, std::string("message %d %d %d %d"), 1, 2, 3, 4)); //This works, but I am not sure if it is correct
- RCLCPP_DEBUG(g_logger, "message %d %d %d %d", 1, 2, 3, 4)); // standard use

but never:

- RCLCPP_DEBUG(g_logger, std::string("message "),  std::string("one"));

Only first argument is possible to be a string. If I am wrong, let me know, because I am not really sure if this can happen...

Reply:
That looks right to me, the documentation as it stands is that the first string is a format string (which may have no format arguments) and the rest of the arguments are input for the format string:

http://docs.ros2.org/bouncy/api/rclcpp/logging_8hpp.html#a44818c8a1fd292cd0e81759e956765d7